lib/checkunusedvar.cpp

```diff
@@ -1609,17 +1609,31 @@ bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func, const To
     }
 
     bool sideEffectReturnFound = false;
+    std::list<const Variable*> pointersToGlobals;
     for (Token* bodyToken = func.functionScope->bodyStart->next(); bodyToken != func.functionScope->bodyEnd;
-         bodyToken = bodyToken->next()) {
+            bodyToken = bodyToken->next())
+    {
+        // check variable inside function body
         const Variable* bodyVariable = bodyToken->variable();
         if (bodyVariable) {
-            // check variable for side-effects
             if (!isVariableWithoutSideEffects(*bodyVariable)) {
                 return false;
             }
             // check if global variable is changed
-            if (bodyVariable->isGlobal()) {
-                return false; // TODO: analyze global variable usage
+            if (bodyVariable->isGlobal() ||
+                (std::find(pointersToGlobals.begin(), pointersToGlobals.end(), bodyVariable) != pointersToGlobals.end()))
```

Comment:
It seems we should optimise the lookup. So please use `std::set` instead for `pointersToGlobals`.